docs/src/content/docs/request-nanostores.mdx

```diff
@@ -0,0 +1,57 @@
+---
+title: Request Scoped Nanostores
+description: Make your Nanostores concurrent safe and shared from server to client
+packageName: '@inox-tools/request-nanostores'
+challenger: [Florian, Oliver]
+---
+
+[Nanostores](https://github.com/nanostores/nanostores) is a great and super compact state management library that is compatible with multiple frameworks, even with multiple at the same time. Nanostores has been Astro's recommended option for sharing state [between components](https://docs.astro.build/en/recipes/sharing-state/) and [between client islands](https://docs.astro.build/en/recipes/sharing-state-islands/).
+
+Using Nanostores directly with Astro has some caveats, as explained on Astro's ["Why Nano Stores?" FAQ](https://docs.astro.build/en/recipes/sharing-state-islands/#why-nano-stores). They are meant for the client-side. Using them on the server, be it on framework components or on the frontmatter of Astro components, may cause problems with data race between requests. Once the page is rendered, the stores on the client won't have the data from the stores on the server, which may cause flickering and flashing of content on the screen as the client renders a different content from the server.
+
+This integration bridges this gap and enables the use of Nanostores in Astro across server and client isolated on each request.
+
+## Installing the dependency
+
+import { PackageManagers } from 'starlight-package-managers';
+
+<PackageManagers type="exec" pkg="astro" args="add @inox-tools/request-nanostores" />
+
+### Prerequisites
+
+When using the Cloudflare adapter, you'll need to [enable AsyncLocalStorage manually](https://developers.cloudflare.com/workers/runtime-apis/nodejs/#enable-only-asynclocalstorage).
+
+## How to use
```

Comment:
You show how to define the atoms, would be great to show how to use them (eg. set atom on the server and read it on the client)

Reply:
I'm not great at frontend (read, absolutely useless 😅). But see if the example I added helps.